v2/pkg/catalog/loader/loader.go

```diff
@@ -195,8 +203,14 @@ func areWorkflowOrTemplatesValid(store *Store, filteredTemplatePaths map[string]
 				areTemplatesValid = false
 			}
 		} else {
+			if old, ok := uniqueTemplatesIDValidateMap[template.ID]; !ok {
+				uniqueTemplatesIDValidateMap[template.ID] = templatePath
+			} else {
+				areTemplatesValid = false
+				gologger.Warning().Msgf("Found duplicate template ID during validation %s => %s: %s\n", templatePath, old, template.ID)
```

Comment:
"Duplicate template ID found during validation" sounds a bit better imo.

Reply:
Since paths can contain spaces, I usually put them between single quotes to improve the readability / not cause confusion to the user.